application/controllers/admin/Tokens.php

```diff
@@ -737,17 +737,16 @@ public function addnew(int $iSurveyId)
      * Edit Tokens
      * @param int $iSurveyId
      * @param int $iTokenId
-     * @param boolean $ajax
      * @return false|null
-     * @todo When is this function used without Ajax?
      */
-    public function edit($iSurveyId, $iTokenId, $ajax = false)
+    public function edit($iSurveyId, $iTokenId)
```

Comment:
Shouldn't we look for the places where this is called to remove a third parameter if appears?

Also, not sure about changing the signature of a public function.
We tried not to do so in the past. But I believe that could be evaluated case by case. As this is a controller, perfectly fine.

Reply:
Here the only usage is to return the partial view or the complete view.

You return the partial view if it's an ajax request, complete if not. Get is ajax request by parameter is an issue.

Maybe deprecate the unused part ?

Reply:
(i can search and remove the place where the ajax is added in url)

Reply:
> (i can search and remove the place where the ajax is added in url)

Done

Reply:
> Also, not sure about changing the signature of a public function. We tried not to do so in the past. But I believe that could be evaluated case by case. As this is a controller, perfectly fine.

Done too, but really deprecate ;) and put version